### PR TITLE
Simplify identifier creation in ConcatenatedModules

### PR DESCRIPTION
**Pull request by lencioni**
**What kind of change does this PR introduce?**

Optimization

**Did you add tests for your changes?**

No

**If relevant, link to documentation update:**

**Summary**

This was implemented in #5997 to fix some memory issues, but I think we
can make this a little more efficient. By using a single forEach instead
of a map, filter, and join, we avoid unnecessary array creations and
iterations.

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

**Comment by sokra** [comment]
I did measure it and calling hash.update often is more expensive than calling it with a large string.

As explanation I would guess `update` calls into node.js C++ + OpenSSL while concatenating strings is very optimized in JIT.

**Comment by sokra** [comment]
But thanks for the PR. We can optimize webpack by doing the reverse: calling hash.update less often and instead concatenate string.

**Comment by lencioni** [comment]
Thanks for doing the profiling @sokra! I've added a commit to this PR that makes this change.

File: lib/optimize/ConcatenatedModule.js [hunk 1]
```diff
@@ -295,12 +295,13 @@ class ConcatenatedModule extends Module {
 	}
 
 	_createIdentifier() {
-		const orderedConcatenationListIdentifiers = this._orderedConcatenationList.map(info => {
-			switch(info.type) {
-				case "concatenated":
-					return info.module.identifier();
+		let orderedConcatenationListIdentifiers;
```
Comment (egoroof) [inline lib/optimize/ConcatenatedModule.js]:
```js
let a;
a += 'wat';
console.log(a); // "undefinedwat"
```

File: lib/optimize/ConcatenatedModule.js [hunk 2]
```diff
@@ -295,12 +295,13 @@ class ConcatenatedModule extends Module {
...
+		for(let i = 0; i < this._orderedConcatenationList.length; i++) {
+			if(this._orderedConcatenationList[i].type === "concatenated") {
+				orderedConcatenationListIdentifiers += this._orderedConcatenationList[i].module.identifier();
+				orderedConcatenationListIdentifiers += " ";
```
Comment (egoroof) [inline lib/optimize/ConcatenatedModule.js]: That's strange

File: lib/optimize/ConcatenatedModule.js (hunk 1 above)
Comment (lencioni) [reply lib/optimize/ConcatenatedModule.js]: lol, woops! fixed now. Thanks!

**Comment by sokra** [comment]
Thanks

**Comment by sokra** [comment]
#5718

**Pull request merged**